### ISSUE TITLE
Fix S1144: False positive for partial methods/classes

### ISSUE THREAD

**Issue by gkello** [SonarSource/sonar-csharp#710]
### Description

In case of partial classes/methods, in case when partial method is declared and used in one file but implemented in other file, the implementation gets false positive S1144 detected.

In case the both class parts are in same C# file the issue is not detected.

### Repro steps

file Partial.generated.cs
```C#
namespace SonarCheck
{
    public partial class Partial
    {
        public void DoWork()
        {
            Initialize();
        }
        partial void Initialize();
    }
}
```

file Partial.custom.cs
```C#
namespace SonarCheck
{
    partial class Partial
    {
        partial void Initialize()
        {
        }
    }
}
```

### Expected behavior

No issues detected

### Actual behavior

For Partial.custom.cs file, an issue is detected:
```
Warning	S1144	Remove the unused private method 'Initialize'.
```
### Related information

* SonarAnalyzer.CSharp 6.3.0.2862
* Visual Studio Professional 2017, version 15.3.1

**Comment by michalb-sonar** [comment]
Thanks for reporting @gkello! I can reproduce the problem, we'll address it.

Interestingly, on my machine issue is reported even if the partial definitions are in the same file.
